### PR TITLE
Use optional chaining to simplify the detection code

### PR DESCRIPTION
**Pull request by fregante**
Mini proposal to simplify the detection code. Feel free to reject it if the drawbacks are too great.

## Pros

- Readable code
- ES2020, yo

## Cons

- It breaks pre-ES2020 build pipelines unless you add a babel transform for it (but I think `babel/preset-env` should already cover it 🤔) and transpile it down

### CODE REVIEW

File: src/browser-polyfill.js [hunk 1]
```diff
@@ -6,11 +6,11 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 "use strict";
 
-if (typeof globalThis != "object" || typeof chrome != "object" || !chrome || !chrome.runtime || !chrome.runtime.id) {
+if (!globalThis.chrome?.runtime?.id) {
   throw new Error("This script should only be loaded in a browser extension.");
 }
 
-if (typeof globalThis.browser === "undefined" || Object.getPrototypeOf(globalThis.browser) !== Object.prototype) {
+if (!globalThis.browser?.runtime?.id || Object.getPrototypeOf(globalThis.browser) !== Object.prototype) {
```
Comment (fregante) [inline src/browser-polyfill.js:13]: Note: this code should be further refined due to:

- https://github.com/mozilla/webextension-polyfill/issues/364

**Comment by fregante** [comment]
> have you already looked into that failure?

I assumed that my second commit would fix that. I'm not sure exactly why it would fail, but in that case this PR doesn't bring a dramatic improvement of the situation. A bugfix for https://github.com/mozilla/webextension-polyfill/issues/364 would have to review the whole condition anyway, so I dropped that part.

**Comment by fregante** [comment]
Will this be merged soon? It's a 1-line change that's been open for 4 months 😅

**Comment by rpl** [comment]
> Will this be merged soon? It's a 1-line change that's been open for 4 months sweat_smile

I'm sorry, for sure merging the PR takes just the press of a button, but before releasing a new version there is also some maintenance work to be done in this repo, and unfortunately I have been quite busy and haven't been able to free myself from other things enough to also do that maintenance work. 

This is mainly a cosmetic change ([based on the compatibility table on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) the browser versions not providing optional chaining are not exactly "recent") and so release it wouldn't change much in practice.

I'll merge it in the meantime, and hopefully get back to this repo for the rest of maintenance work soon enough.

**Comment by fregante** [comment]
I see Babel Preset Env in the dependencies, was it not applied? I assumed code went through it to avoid mistakenly changing the output target.

**Comment by fregante** [comment]
As for your issue specifically, updating your modules will fix it. This package version was updated correctly, each 0.x.0 version *can* be semver breaking.

**Comment by davidbielik** [comment]
Thanks @fregante i'm positive it's an issue on my end, just didn't want to have to start messing with webpack and npm dependencies this for the one-line change 🤣

**Comment by PoziWorld** [comment]
Would the team consider transpiling it to ES5 for CDN-hosted versions?

I develop extensions that are also used by users of Chrome v49 on Windows XP and Firefox v56/Waterfox.

See https://github.com/mozilla/webextension-polyfill/issues/420.

**Comment by fregante** [comment]
> for CDN-hosted versions

You can't load scripts from CDN in extensions



> I develop extensions that are also used by users of Chrome v49 on Windows XP and Firefox v56/Waterfox.

You're likely using babel in your build, so you can include the polyfill in the transpilation — or keep using v0.9.0. There haven't been any real changes in this polyfill for the past 2+ years

**Pull request merged**